codebase2/codebase-sqlite/U/Codebase/Sqlite/Operations.hs

```diff
@@ -1130,11 +1134,71 @@ rootNamesByPath path = do
     convertTerms = fmap (bimap s2cTextReferent (fmap s2cConstructorType))
     convertTypes = fmap s2cTextReference
 
+-- | NOTE: requires that the codebase has an up-to-date name lookup index. As of writing, this
+-- is only true on Share.
+--
+-- Get the list of a names for a given Referent.
+termNamesForRefWithinNamespace :: BranchHash -> Q.NamespaceText -> C.Referent -> Maybe S.ReversedSegments -> Transaction [S.ReversedSegments]
```

Comment:
These area all just wrappers for converting between application and database types.